src/python/pants/backend/docker/goals/run_image.py

```diff
@@ -49,8 +53,16 @@ async def docker_image_run_request(
         Get(EnvironmentVars, EnvironmentVarsRequest(options_env_aware.env_vars)),
         Get(BuiltPackage, PackageFieldSet, build_request),
     )
+
+    docker_run_args = options.run_args + (
+        wrapped_target.target.get(DockerImageRunExtraArgsField).value or ()
+    )
```

Comment:
Perhaps nit, but I feel it would be more inline with how field sets are designed to work, to have a `field_set.get_run_args(options)` here:
```suggestion
    docker_run_args = field_set.get_run_args(options)
```

Which could be implemented as:
```python
@dataclass(frozen=True)
class DockerRunFieldSet(RunFieldSet):
    required_fields = (DockerImageSourceField,)
    run_in_sandbox_behavior = RunInSandboxBehavior.RUN_REQUEST_HERMETIC

    extra_run_args: DockerImageRunExtraArgsField

    def get_run_args(self, options: DockerOptions) -> tuple[str, ...]:
        return tuple(options.run_args + self.extra_run_args.value)
```